CHANGELOG.md

```diff
@@ -2,6 +2,17 @@
 
 ## Unreleased
 
+### General
```

Comment:
Regarding the changelog structure, I'd like to keep the main categories as "Breaking changes" / "Additions" / "Deprecations" / "Bugfixes", so that users can quickly guess their upgrade path from the changelog.

Thematic sub-categories within that are a nice idea, though imo all the entries you added would fit in the "Clients & Protocols" sub-category.